nodejs/week1/mealSharingWebsite/src/backend/index.js

```diff
@@ -0,0 +1,64 @@
+const express = require("express");
+const fs = require("fs");
+const app = express();
+const meals = JSON.parse(fs.readFileSync(__dirname + "/data/meals.json"));
+const reviews = JSON.parse(fs.readFileSync(__dirname + "/data/reviews.json"));
+const reservations = JSON.parse(fs.readFileSync(__dirname + "/data/reservations.json"));
+
+//Error handling
+//Middleware
+
+app.get('/', (req, res, next) => {
+    res.status(404);
+    res.send('Error no page found');
+    next()
+});
+
+//Setting of routes
+//Respond with the json for all the meals. For each meal, if there are reviews matching it's meal ID, add these reviews to each meal in the form of an array. 
+
+app.get("/meals", (req, res, next) => {
+    //res.json(meals)
+    meals.forEach(meal => {
+        meal.reviews = [];
+        for (let review of reviews) {
+            if (meal.id === review.mealId) {
+                meal.reviews.push(review);
+            }
+        };
```

Comment:
Really good work. There is a bug here though. Since you are appending reviews to meals in the callback function for `meals` route, your app seems to work as expected if the first route the user requests is `meals`. However, if the user requests `cheap meals`, `large meals`, or `random meal` without having first visited the `meals` route, the meals returned on these routes do not have any reviews attached to them.
Can you see how this can be fixed?

Reply:
Thanks for your review and comment , will check and update it soon.